### PR TITLE
Make null primitive value return None

### PR DESCRIPTION
**Pull request by tonykwok1992**
Currently selecting rows with null value in a nullable int/long columns will gives Some(0) even when the schema object field has type Option[Int], due to the fact that resultSet.getInt(columnIndex) gives 0 for null value rather than null,

### CODE REVIEW

File: postgres/src/test/scala/zio/sql/postgresql/SelectSpec.scala [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    test("Select null int value as None Option") {
+      val movieColumns = MoviesSchema.id ++ MoviesSchema.rating
+      val selectStmt = select(movieColumns).from(MoviesSchema.movies).where(MoviesSchema.id === 1)
+      val selectResult = execute(selectStmt.to((MoviesSchema.Movies.apply _).tupled)).runCollect
+
+      val insertAssertion = for {
+        movies <- selectResult
+      } yield assert(movies.toList)(equalTo(List(MoviesSchema.Movies(
+        id = 1,
+        rating = None
```
Comment (tonykwok1992) [inline postgres/src/test/scala/zio/sql/postgresql/SelectSpec.scala:20]: This is Some(0) before the change

**Comment by tonykwok1992** [comment]
Hello, can I get a review on this? thanks!

File: postgres/src/test/scala/zio/sql/postgresql/SelectSpec.scala [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package zio.sql.postgresql
+
+import zio.Cause
+import zio.test.Assertion._
+import zio.test._
+
+object SelectSpec extends PostgresRunnableSpec with DbSchema {
```
Comment (sviezypan) [inline postgres/src/test/scala/zio/sql/postgresql/SelectSpec.scala:7]: would you move this to PostgresSqlModuleSpec? as there are also tests for selects

File: postgres/src/test/scala/zio/sql/postgresql/SelectSpec.scala [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+          equalTo(
+            List(
+              MoviesSchema.Movies(
+                id = 1,
+                rating = None
+              )
+            )
+          )
+        )
+      insertAssertion.mapErrorCause(cause => Cause.stackless(cause.untraced))
```
Comment (sviezypan) [inline postgres/src/test/scala/zio/sql/postgresql/SelectSpec.scala:28]: I think it would work if you just return the for comprehension directly?

**Review by sviezypan — CHANGES_REQUESTED** [review]
hi @tonykwok1992 thank you for this work, this is great catch and fix. Just 2 minor comments and then its good to be merged.

**Pull request merged**